src/lib/components/BaseModalElement/index.js

```diff
@@ -0,0 +1,176 @@
+    this.manageFocus = this.manageFocus.bind(this);
+    this.getTrigger = this.getTrigger.bind(this);
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.addEventListener("click", this.onClick);
+    // Set tabindex to -1 so modal can be focused when it's opened.
+    this.tabIndex = -1;
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    this.removeEventListener("click", this.onClick);
+  }
+
+  set open(val) {
+    if (this.open_ === val) return;
+
+    const oldVal = this.open_;
+
+    this.open_ = val;
+    if (this.open_) {
+      // Must get trigger before manipulating the DOM.
+      this.getTrigger();
+      document.addEventListener("keyup", this.onKeyUp);
+      window.addEventListener("resize", this.onResize);
+    } else {
+      window.removeEventListener("resize", this.onResize);
+    }
+
+    this.manageDocument();
+    this.animatable = true;
+    this.addEventListener("animationend", this.onAnimationEnd, {
+      once: true,
+    });
+
+    this.requestUpdate("open", oldVal);
+  }
+
+  get open() {
+    return this.open_;
+  }
+
+  onClick(e) {
+    // Check whether the event target matches the currentTarget
+    // so the modal only closes when a user clicks the scrim.
+    if (e.currentTarget !== e.target) return;
+    this.open = false;
+  }
+
+  onKeyUp(e) {
+    // Close modal when user presses Escape.
+    if (e.key === "Escape") {
+      this.open = false;
+      document.removeEventListener("keyup", this.onKeyUp);
+    }
+  }
+
+  // Assumes modal on/off transitions are handled by an animation
+  // rather than a transition.
+  onAnimationEnd() {
+    this.animatable = false;
+    // Must wait until animation ends to change focus and check overflow.
+    this.manageFocus();
+    if (this.open) this.onResize();
+    this.inert = !this.open;
+  }
+
+  onResize() {
+    // Apply a class to the modal if it has overflow to allow for styling changes
+    // (e.g., adding borders to the child element handling overflow).
+    // If the client component needs to use a different class for the element
+    // handling overflow, it will need its own animationend listener.
+    // See the Assessment component for an example.
+    const content = this.querySelector(".web-modal__content");
+
+    if (!content) return;
+    handleOverflow(content, "height", "web-modal--overflow", this);
+  }
+
+  getTrigger() {
+    // When the modal is opened, asign a class to the triggering element
+    // so it can be refocused when the modal is closed.
+    const trigger = document.activeElement;
+
+    trigger.classList.add("js-modal-trigger");
+  }
+
+  manageDocument() {
+    // Inert everything other than the modal when it's open.
+    const main = document.querySelector("main");
+    const header = document.querySelector("web-header");
+    const footer = document.querySelector("footer");
+    const overFlowClass = "web-modal__overflow-hidden";
+
+    if (this.open) {
```

Comment:
So this stuff should be on if _any_ modal is open (and it might be a bad idea to allow multiple modals to be open, too). So there needs to be some static singleton state here.

Reply:
I added an action but didn't add logic to prevent multiple modals at once because we already have a case where we need that: the assessment is modal on mobile, as is the question feedback form. I know nested modals sounds gross in the abstract, but when you [try it out](https://codepen.io/mfriesenhahn/full/mdygdej), I think it ends up being the least-bad option.